retriever/engines/mysql.py

```diff
@@ -39,6 +41,26 @@ class engine(Engine):
                       "{db}.{table}"),
                      ]
 
+    def check_credentials(self):
+        """
+        Checks the credentials of the user in the file path ~/.my.cnf for the MySQL
+        credentials by checking the username provided by the user in cli.
+        If found, it updates the credentials to initialize the connection, else uses the
+        default ones.
+        """
+        self.get_input()
+        if os.path.exists(MYSQL_CONF_PATH):
+            sql_conf = open_fr(MYSQL_CONF_PATH)
+            sql_conf_lines = sql_conf.readlines()
+            for line in sql_conf_lines:
+                if re.search("user = ", line) and re.sub("user = ", "", line) == self.opts["user"]:
```

Comment:
This line `re.search("user = ", line)` passes for files that have `user = ` with spaces but will not pass for a line like "user= "
So we have to compile a regex that matches all the possible arrangements like the sample below



```
[client]
user=mysqluser
password=mysqlpass

```


```
[client]
user     =     mysqluser

password  =mysqlpass

```

Something else to consider, some files have settings for different groups.
I think our script should look for the default client group, which is [client] by default.


In order to design this, We can start processing line only after we hit the line containing [client]
and end the processing if any line start with [

To handle this we can set flags to denote when we start processing, i.e line contains [client]
and set a flag that will stop the processing when we hit the line starting with []  

In this way we avoid taking values from other groups

```

[mysqladmin]

user =admin4

[client]

user = retriever
password = testpass
interactive_timeout=400

[mysql]

[mysqladmin]

```